### PR TITLE
fixing dls/fls logic around numeric aggregations

### PR DESCRIPTION
**Pull request by kaituo**
Signed-off-by: Kaituo Li <kaituo@amazon.com>

### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/788

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by amitgalitz** [comment]
Weird that it says it can't get the zip for job scheduler but even the link in the error message works fine to download job-scheduler-1.3.8.0

**Comment by kaituo** [comment]
> Weird that it says it can't get the zip for job scheduler but even the link in the error message works fine to download job-scheduler-1.3.8.0

Indeed.

**Review by amitgalitz — APPROVED** [review]

File: src/main/java/org/opensearch/ad/AnomalyDetectorJobRunner.java [hunk 1]
```diff
@@ -218,28 +218,11 @@ protected void runAdJob(
...
-            // It's possible that user create domain with security disabled, then enable security
-            // after upgrading. This is for BWC, for old detectors which created when security
-            // disabled, the user will be null.
-            user = "";
-            roles = settings.getAsList("", ImmutableList.of("all_access", "AmazonES_all_access"));
-        } else {
-            user = jobParameter.getUser().getName();
-            roles = jobParameter.getUser().getRoles();
-        }
+        User userInfo = SecurityUtil.getUserFromJob(jobParameter, settings);
```
Comment (ylwu-amzn) [inline src/main/java/org/opensearch/ad/AnomalyDetectorJobRunner.java:221]: Is this ready user information stored in AD job? I think we have no way to get user information from security plugin. Just asking to learn more

File: src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java [hunk 2]
```diff
@@ -156,10 +156,7 @@
 import org.opensearch.ad.transport.handler.AnomalyIndexHandler;
 import org.opensearch.ad.transport.handler.AnomalyResultBulkIndexHandler;
 import org.opensearch.ad.transport.handler.MultiEntityResultHandler;
-import org.opensearch.ad.util.ClientUtil;
-import org.opensearch.ad.util.DiscoveryNodeFilterer;
-import org.opensearch.ad.util.IndexUtils;
-import org.opensearch.ad.util.Throttler;
+import org.opensearch.ad.util.*;
```
Comment (ylwu-amzn) [inline src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java:159]: minor: should we avoid using import *?

**Review by ylwu-amzn — APPROVED** [review]

File: src/main/java/org/opensearch/ad/AnomalyDetectorJobRunner.java (hunk 1 above)
Comment (kaituo) [reply src/main/java/org/opensearch/ad/AnomalyDetectorJobRunner.java:221]: Yes, it is from AD job.

File: src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java (hunk 2 above)
Comment (kaituo) [reply src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java:159]: Yes, will fix in other PRs since 1.3.8 is due today.

**Pull request merged**